### PR TITLE
feat: use nix w/ poetry

### PR DESCRIPTION
**Pull request by andrewgazelka**

### CODE REVIEW

**Comment by andrewgazelka** [comment]
* **#2** <a href="https://app.graphite.dev/github/pr/andrewgazelka/discord-code-display/2?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/andrewgazelka/discord-code-display/2?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: flake.nix [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  };
+
+  # has standard of which keys
+  outputs = {self, ...} @ inputs:
+    inputs.flake-utils.lib.eachDefaultSystem (system: let
+      pkgs = import inputs.nixpkgs {localSystem = {inherit system;};};
+      inherit (pkgs) lib;
+
+      # Project metadata
+      pyproject = builtins.fromTOML (builtins.readFile ./pyproject.toml);
```
Comment (x10an14) [inline flake.nix]: Does this file exist?

File: flake.nix [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        mkPoetryEnv
+        ;
+
+      # Hardcode python version:
+      python = pkgs.python312;
+      poetryConfig = {
+        inherit python;
+        projectDir = self;
+        preferWheels = true;
+        groups = ["rest" "bigquery_search"];
```
Comment (x10an14) [inline flake.nix]: Does the `pyproject.toml` file contain these pip groups?

File: flake.nix [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+      };
+      pythonApp = pkgs.writeShellApplication {
+        inherit (pyproject.tool.poetry) name;
+        runtimeInputs = [
+          # The actual Poetry app:
+          (mkPoetryApplication poetryConfig)
+          .dependencyEnv
+        ];
+        text = "uvicorn nks_ds.rest.main:app --host 0.0.0.0 --port 8080";
```
Comment (x10an14) [inline flake.nix]: This will fail at run-time (not build time) unless your app does/supports the same thing...

File: flake.nix [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+      };
+      poetryDevEnv = mkPoetryEnv (lib.updateManyAttrsByPath [
+          {
+            path = ["groups"];
+            update = old: old ++ ["dev" "lint"];
```
Comment (x10an14) [inline flake.nix]: Does the `pyproject.toml` file contain these pip groups?

**Pull request closed without merging**